pyfixest/report/summarize.py

```diff
@@ -293,30 +328,62 @@ def etable(
         print(res_all)
-        if signif_code:
-            print(
-                f"Significance levels: * p < {signif_code[2]}, ** p < {signif_code[1]}, *** p < {signif_code[0]}"
-            )
-            print(f"Format of coefficient cell:\n{coef_fmt_title}")
-        return None
+        # Generate notes string if user has not provided any 
+        if notes is None:
+            if signif_code:
+                print(
+                    f"Significance levels: * p < {signif_code[2]}, ** p < {signif_code[1]}, *** p < {signif_code[0]}"
+                )
+                print(f"Format of coefficient cell: {coef_fmt_title}")
+            return None
+        else:
+            print(notes)
+            return None
     elif type in ["df", "tex"]:
-        res_all = _tabulate_etable_df(res_all, n_fixef, caption)
+        # Prepare Multiindex for columns 
+        id_dep = [""] + dep_var_list                                # depvars
+        if model_heads is None:
+            id_head = [""] * (1+len(models))
+        else:
+            id_head = [""] + model_heads                            # model_heads provided by user
+        id_num = [""]+[f"({s})" for s in range(1, len(models) + 1)] # model numbers
+
+        res_all = pd.concat([res, nobs_fixef_df], ignore_index=True)
+        # When no depvars & headlines should be displayed then use simple index
+        # otherwise generate MultiIndex & determine order of index levels as specified by head_order 
+        if head_order=="":
+            res_all.columns = pd.Index(id_num)
+        else:
+            cindex = [ {'h': id_head, 'd': id_dep}[c] for c in head_order] + [id_num]
+            res_all.columns= pd.MultiIndex.from_arrays(cindex)
+        
         if type == "df":
+            # Generate notes string if user has not provided any 
+            if notes is None:
+                notes = (
+                    f"Significance levels: * p < {signif_code[2]}, ** p < {signif_code[1]}, *** p < {signif_code[0]}. "
+                    + f"Format of coefficient cell:\n{coef_fmt_title}"
+                )
+            res_all = _tabulate_etable_df(df=res_all, n_fixef=n_fixef, notes=notes)
             return res_all
-        else:
-            return res_all.to_latex()
+        elif type == "tex":
+            # Generate notes string if user has not provided any 
+            if notes is None:
+                notes = (
+                    f"Significance levels: $*$ p $<$ {signif_code[2]}, $**$ p $<$ {signif_code[1]}, $***$ p $<$ {signif_code[0]}. "
+                    + f"Format of coefficient cell: {coef_fmt_title}"
+                )
+            latex_res=_tabulate_etable_tex(df=res_all, n_coef=res.shape[0], n_fixef=n_fixef, n_models=len(models), notes=notes)
+            if filename is not None:
+                with open(filename, "w") as f:
+                    f.write(latex_res)  # Write the latex code to a file
+            return latex_res 
```

Comment:
Instead of ending with `return latex_res`, should we not have 

```python
print(latex_res)
```
? This way, we'd directly output nicely formatted latex that can directly be pasted into the tex document. 

```python
tex = pf.etable(
    [fit1, fit2, fit3, fit4, fit5],
    type = "tex"
)

print(tex)
```
returns

\renewcommand\cellalign{t}
\begin{threeparttable}
\begin{tabular}{lccccc}
\toprule
 & \multicolumn{5}{c}{Y} \\
\cmidrule(lr){2-6} 
 & (1) & (2) & (3) & (4) & (5) \\
\midrule
X1 & \makecell{-0.949*** \\ (0.069)} & \makecell{-0.919*** \\ (0.065)} & \makecell{-0.950*** \\ (0.067)} & \makecell{-0.924*** \\ (0.061)} & \makecell{-0.924*** \\ (0.061)} \\
X2 &  &  & \makecell{-0.174*** \\ (0.018)} & \makecell{-0.174*** \\ (0.015)} & \makecell{-0.185*** \\ (0.025)} \\
X1:X2 &  &  &  &  & \makecell{0.011 \\ (0.018)} \\
\midrule
f1 & x & x & x & x & x \\
f2 & - & x & - & x & x \\
\midrule
$R^2$ & 0.437 & 0.609 & 0.489 & 0.659 & 0.659 \\
S.E. type & by: f1 & by: f1 & by: f1 & by: f1 & by: f1 \\
Observations & 997 & 997 & 997 & 997 & 997 \\
\bottomrule
\end{tabular}
\footnotesize Significance levels: $*$ p $<$ 0.05, $**$ p $<$ 0.01, $***$ p $<$ 0.001. Format of coefficient cell: Coefficient 
 (Std. Error)
\end{threeparttable}

while `tex` is a python string.

```
# '\\renewcommand\\cellalign{t}\n\\begin{threeparttable}\n\\begin{tabular}{lccccc}\n\\toprule\n & \\multicolumn{5}{c}{Y} \\\\\n\\cmidrule(lr){2-6} \n & (1) & (2) & (3) & (4) & (5) \\\\\n\\midrule\nX1 & \\makecell{-0.949*** \\\\ (0.069)} & \\makecell{-0.919*** \\\\ (0.065)} & \\makecell{-0.950*** \\\\ (0.067)} & \\makecell{-0.924*** \\\\ (0.061)} & \\makecell{-0.924*** \\\\ (0.061)} \\\\\nX2 &  &  & \\makecell{-0.174*** \\\\ (0.018)} & \\makecell{-0.174*** \\\\ (0.015)} & \\makecell{-0.185*** \\\\ (0.025)} \\\\\nX1:X2 &  &  &  &  & \\makecell{0.011 \\\\ (0.018)} \\\\\n\\midrule\nf1 & x & x & x & x & x \\\\\nf2 & - & x & - & x & x \\\\\n\\midrule\n$R^2$ & 0.437 & 0.609 & 0.489 & 0.659 & 0.659 \\\\\nS.E. type & by: f1 & by: f1 & by: f1 & by: f1 & by: f1 \\\\\nObservations & 997 & 997 & 997 & 997 & 997 \\\\\n\\bottomrule\n\\end{tabular}\n\\footnotesize Significance levels: $*$ p $<$ 0.05, $**$ p $<$ 0.01, $***$ p $<$ 0.001. Format of coefficient cell: Coefficient \n (Std. Error)\n\\end{threeparttable}'
```